girder/cli/build.py

```diff
@@ -63,6 +63,19 @@ def main(dev, mode, watch, watch_plugin, npm, reinstall):
         npmLockFile = os.path.join(staging, 'package-lock.json')
         if os.path.exists(npmLockFile):
             os.unlink(npmLockFile)
+
+        # Remove any lingering node_modules to ensure clean install
+        plugin_dirs = [path.replace('file:', '') for path in _collectPluginDependencies().values()]
```

Comment:
`_collectPluginDependencies` is expensive to run, since it ultimately iterates over Python entry points. If possible, it'd be nice to only run it once throughout `girder build`, but it's not a big deal.